osu.Game.Tournament/Screens/StablePathSelectScreen.cs

```diff
@@ -0,0 +1,203 @@
+                                        Font = OsuFont.Default.With(size: 40)
+                                    },
+                                },
+                                new Drawable[]
+                                {
+                                    directorySelector = new DirectorySelector(initialPath)
+                                    {
+                                        RelativeSizeAxes = Axes.Both,
+                                    }
+                                },
+                                new Drawable[]
+                                {
+                                    new FillFlowContainer
+                                    {
+                                        Anchor = Anchor.Centre,
+                                        Origin = Anchor.Centre,
+                                        Direction = FillDirection.Horizontal,
+                                        Spacing = new Vector2(20),
+                                        Children = new Drawable[]
+                                        {
+                                            new TriangleButton
+                                            {
+                                                Anchor = Anchor.Centre,
+                                                Origin = Anchor.Centre,
+                                                Width = 300,
+                                                Text = "Select stable path",
+                                                Action = () => changePath(storage)
+                                            },
+                                            new TriangleButton
+                                            {
+                                                Anchor = Anchor.Centre,
+                                                Origin = Anchor.Centre,
+                                                Width = 300,
+                                                Text = "Auto detect",
+                                                Action = autoDetect
+                                            },
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                },
+                new BackButton
+                {
+                    Anchor = Anchor.BottomLeft,
+                    Origin = Anchor.BottomLeft,
+                    State = { Value = Visibility.Visible },
+                    Action = () => sceneManager?.SetScreen(typeof(SetupScreen))
+                }
+            });
+        }
+
+        private void changePath(Storage storage)
+        {
+            var target = directorySelector.CurrentDirectory.Value.FullName;
+            Logger.Log($"Changing Stable CE location to {target}");
+
+            if (File.Exists(Path.Combine(target, "ipc.txt")))
```

Comment:
I'd invert and deal with the negative part first (`ipc.txt` not found), and early-return if it didn't succeed.

This is also a piece of code (albeit a one-liner) that has duplicated filenames (see `FileBasedIPC.checkExists`). Would like to see at least the filename extracted to a common constant.